trade-finance-manager-ui/server/controllers/case/parties/index.js

```diff
@@ -39,87 +47,304 @@ const getCaseParties = async (req, res) => {
   });
 };
 
-const getPartyDetails = (partyType) => (
-  async (req, res) => {
-    const { user } = req.session;
+/**
+ * Renders party specific URN edit page
+ * @param {Object} req Express request
+ * @param {Object} res Express response
+ * @returns {Object} Express response as rendered party page.
+ */
+const getPartyDetails = async (req, res) => {
+  if (!req) {
+    console.error('Invalid request received');
+    return res.redirect('/not-found');
+  }
 
-    const canEdit = userCanEdit(user);
+  const { user } = req.session;
+  const party = partyType(req.url);
 
-    if (!canEdit) {
-      return res.redirect('/not-found');
-    }
+  const canEdit = userCanEdit(user);
 
-    const dealId = req.params._id;
-    const deal = await api.getDeal(dealId);
+  if (!canEdit) {
+    console.error('Invalid user privilege.');
+    return res.redirect('/not-found');
+  }
 
-    if (!deal) {
-      return res.redirect('/not-found');
-    }
+  const dealId = req.params._id;
+  const deal = await api.getDeal(dealId);
 
-    return res.render(`case/parties/edit/${partyType}-edit.njk`, {
-      userCanEdit: canEdit,
-      renderEditLink: false,
-      renderEditForm: true,
-      activePrimaryNavigation: 'manage work',
-      activeSubNavigation: 'parties',
-      deal: deal.dealSnapshot,
-      tfm: deal.tfm,
-      dealId,
-      user: req.session.user,
-      urn: deal.tfm.parties[partyType].partyUrn,
-    });
+  if (!deal) {
+    console.error('Invalid deal.');
+    return res.redirect('/not-found');
   }
-);
 
-const getExporterPartyDetails = getPartyDetails('exporter');
-const getBuyerPartyDetails = getPartyDetails('buyer');
-const getAgentPartyDetails = getPartyDetails('agent');
-const getIndemnifierPartyDetails = getPartyDetails('indemnifier');
+  if (!party) {
+    console.error('Invalid party type specified.');
+    return res.redirect('/not-found');
+  }
+
+  return res.render(`case/parties/edit/${party}.njk`, {
+    userCanEdit: canEdit,
+    renderEditLink: false,
+    renderEditForm: true,
+    activePrimaryNavigation: 'manage work',
+    activeSubNavigation: 'parties',
+    deal: deal.dealSnapshot,
+    tfm: deal.tfm,
+    dealId,
+    user: req.session.user,
+    urn: deal.tfm.parties[party].partyUrn,
```

Comment:
is there a check to see if deal.tfm.parties exists?

Reply:
Have added, good shout.